### PR TITLE
chore(repo): dependency reviewer github action

### PR DESCRIPTION
**Pull request by fjnoyp**
Issue #, if available:

Internal Ticket to check dependency vulnerabilities

Description of changes:

Github Action to run on every pull request that checks for any introduced vulnerabilities.

For more details see:
https://github.com/actions/dependency-review-action

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by khatruong2009 — APPROVED** [review]

File: .github/dependency-review-config.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+allow-licenses:
+  - '0BSD'
+  - 'Apache-2.0'
+  - 'BSL-1.0'
```
Comment (haverchuck) [inline .github/dependency-review-config.yml:4]: Let's remove `BSL-1.0`

File: .github/dependency-review-config.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  - 'BSD-1-Clause'
+  - 'BSD-2-Clause-FreeBSD'
+  - 'BSD-2-Clause'
+  - 'BSD-3-Clause-Attribution'
+  - 'BSD-3-Clause'
+  - 'BSD-Source-Code'
+  - 'bzip2-1.0.6'
+  - 'CC-BY-3.0'
+  - 'CC-BY-4.0'
```
Comment (haverchuck) [inline .github/dependency-review-config.yml:13]: Let's remove the two `CC-BY` licenses.

File: .github/dependency-review-config.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  - 'CC0-1.0'
+  - 'curl'
+  - 'ISC'
+  - 'JSON'
+  - 'MIT'
+  - 'MPL-2.0'
```
Comment (haverchuck) [inline .github/dependency-review-config.yml:19]: Remove `MPL-2.0` for now.

File: .github/dependency-review-config.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  - 'NTP'
+  - 'OFL-1.0'
+  - 'OLDAP-2.8'
+  - 'OpenSSL'
+  - 'PDDL-1.0'
```
Comment (haverchuck) [inline .github/dependency-review-config.yml:24]: Remove PDDL-1.0

**Review by haverchuck — APPROVED** [review]
🚢 ship it

**Review by khatruong2009 — APPROVED** [review]

**Pull request merged**